src/ai/backend/manager/exceptions.py

```diff
@@ -25,6 +25,26 @@ class InvalidArgument(Exception):
     pass
 
 
+class RPCError(RuntimeError):
+    """
+    An exception class to represent any error caused in RPC functions.
+    """
+
```

Comment:
The idea to wrap AuthenticationError with a more meaningful `RPCError` looks great. However this error should contain more information about the cause (think of the facts you've mentioned at the PR description!). Please update the implementation to contain more details.